modules/bibupload/lib/bibupload.py

```diff
@@ -2274,11 +2276,11 @@ def _process_document_moreinfos(more_infos, docname, version, docformat, mode):
 
 ### Update functions
 
-def update_bibrec_date(now, bibrec_id, insert_mode_p, pretend=False):
+def update_bibrec_date(now, bibrec_id, insert_mode_p, original_005=None, pretend=False):
     """Update the date of the record in bibrec table """
     if insert_mode_p:
-        query = """UPDATE bibrec SET creation_date=%s, modification_date=%s WHERE id=%s"""
-        params = (now, now, bibrec_id)
+        query = """UPDATE bibrec SET creation_date=%s, modification_date=%s, earliest_date=%s WHERE id=%s"""
+        params = (original_005, now, now, bibrec_id)
     else:
```

Comment:
Great, #1604 is solved in this way.  I fully agree on doing it only in the insert mode.